airbyte-integrations/connectors/destination-mssql-v2/src/test-integration/kotlin/io/airbyte/integrations/destination/mssql/v2/MSSQLWriterTest.kt

```diff
@@ -109,40 +128,138 @@ class MSSQLDataDumper : DestinationDataDumper {
 
     override fun dumpFile(
         spec: ConfigurationSpecification,
-        stream: DestinationStream
+        stream: DestinationStream,
     ): List<String> {
         throw UnsupportedOperationException("destination-mssql doesn't support file transfer")
     }
+}
 
-    private fun getConfiguration(
-        spec: ConfigurationSpecification,
-        stream: DestinationStream
-    ): MSSQLConfiguration {
-        /*
-         * Replace the host, port and schema to match what is exposed
-         * by the container and generated by the test suite in the case of the schema name
-         */
-        val configOverrides =
-            mutableMapOf("host" to MSSQLContainerHelper.getHost()).apply {
-                MSSQLContainerHelper.getPort()?.let { port -> put("port", port.toString()) }
-                stream.descriptor.namespace?.let { schema -> put("schema", schema) }
+class MSSQLDataCleaner(
+    private val shouldCleanUp: Boolean,
+    private val mssqlSpecification: MSSQLSpecification?,
+    private val configProvider: (MSSQLSpecification) -> MSSQLConfiguration
+) : DestinationCleaner {
+
+    private val dateFormatter = DateTimeFormatter.ofPattern("yyyyMMdd")
+
+    override fun cleanup() {
+        if (!shouldCleanUp || mssqlSpecification == null) return
+
+        // Cleanup azure blobs if in BulkLoad configuration
+        cleanupAzureBlobsIfNeeded()
+
+        // Cleanup older test schemas
+        cleanupOldTestSchemas()
+    }
+
+    /** Cleans up blobs older than 1 hour if the load configuration is [BulkLoadConfiguration]. */
+    private fun cleanupAzureBlobsIfNeeded() {
```

Comment:
does azure blob have an equivalent to s3/gcs blob lifecycle policy? might be easier than rolling it ourselves

Reply:
Configured one!
```
{
  "rules": [
    {
      "enabled": true,
      "name": "integration-test-cleanup",
      "type": "Lifecycle",
      "definition": {
        "actions": {
          "baseBlob": {
            "delete": {
              "daysAfterModificationGreaterThan": 2
            }
          }
        },
        "filters": {
          "blobTypes": [
            "blockBlob"
          ],
          "prefixMatch": [
            "mssql-destination-bulk-upload"
          ]
        }
      }
    }
  ]
}
```